### ISSUE TITLE
attiny ADC With internal 2.56v bug

### ISSUE THREAD

**Issue by mostafahk** [SpenceKonde/ATTinyCore#587]
at analogRead() function following line is incorrect because REFS2 is in the ADCSRB register.

``` c
  #if defined(REFS2)
  ADMUX |= (((analog_reference & 0x04) >> 2) << REFS2); //some have an extra reference bit in a weird position.
  #endif
```

please change it to:
``` c
#if defined(REFS2)
ADCSRB |= (((analog_reference & 0x04) >> 2) << REFS2); //some have an extra reference bit in a weird position.
#endif
```

**Comment by SpenceKonde** [comment]
On which part? The REFS bits beyond 0/1 are all over the place/

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by mostafahk** [comment]
I changed the function to the following code and my problem solved. REFS0/1 bits are in the ADMUX but REFS2 exists in the ADCSRB.
\framework-arduino-avr-attiny\cores\tiny\wiring_analog.c


``` c
int analogRead(uint8_t pin)
{
  pin &=127; //strip off the high bit of the A# constants
  #ifndef ADCSRA
  return digitalRead(analogInputToDigitalPin(pin)) ? 1023 : 0; //No ADC, so read as a digital pin instead.
  #endif

  #if defined(ADMUX)
  ADMUX = ((analog_reference & ADMUX_REFS_MASK) << REFS0) | ((pin & ADMUX_MUX_MASK) << MUX0); //select the channel and reference
  #if defined(REFS2)
  ADCSRB |= (((analog_reference & 0x04) >> 2) << REFS2); //some have an extra reference bit in a weird position.
  #endif
  #endif

  #if defined(HAVE_ADC) && HAVE_ADC
  sbi(ADCSRA, ADSC); //Start conversion

  while(ADCSRA & (1<<ADSC)); //Wait for conversion to complete.

  uint8_t low = ADCL;
  uint8_t high = ADCH;
  return (high << 8) | low;
  #else
  return LOW;
  #endif
}
```

The current code doesn't work for the last two rows of voltage reference:
![image](https://user-images.githubusercontent.com/30733372/123503126-1eff7b00-d666-11eb-9e24-df68ee48323c.png)

**Comment by SpenceKonde** [comment]
REFS2 location is different depending on which part you are using, this core supports over a dozen parts. On parts with the third REFS bit, they just shoved it wherever they had an extra bit in an ADC related register. So I ask you AGAIN

WHICH ATTINY PART DO I NEED TO CHANGE THIS FOR?

**Comment by mostafahk** [comment]
ATTINY861

**Comment by SpenceKonde** [comment]
Thank you! I will be sure that is corrected in 2.0.0!
